rascaline-c-api/tests/c-api/logging.cpp

```diff
@@ -0,0 +1,76 @@
+#include <vector>
+#include <string>
+
+#include "rascaline.h"
+#include "catch.hpp"
+#include "helpers.hpp"
+
+static std::vector<std::tuple<int, std::string>> RECORDED_LOG_EVENTS;
+
+static void run_calculation(const char* hypers) {
+    auto* descriptor = rascal_descriptor();
+    REQUIRE(descriptor != nullptr);
+    auto* calculator = rascal_calculator("dummy_calculator", hypers);
+    REQUIRE(calculator != nullptr);
+    auto system = simple_system();
+    auto options = rascal_calculation_options_t {
+        /* use_native_system */ false,
+        /* selected_samples */ nullptr,
+        /* selected_samples_count */ 0,
+        /* selected_features */ nullptr,
+        /* selected_features_count */ 0,
+    };
+
+    CHECK_SUCCESS(rascal_calculator_compute(
+        calculator, descriptor, &system, 1, options
+    ));
+
+    rascal_calculator_free(calculator);
+    rascal_descriptor_free(descriptor);
+}
+
+TEST_CASE("Logging") {
+    auto record_log_events = [](int level, const char* message) {
```

Comment:
weird when I tried a lambda function, it did not work because of type problems

Reply:
Capturing lambda do not work, so the recorded events still need to be a global static variable. Non-capturing lambdas can be used as function pointers.